org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/rename/LSPRenameProcessor.java

```diff
@@ -102,19 +105,22 @@ public RefactoringStatus checkInitialConditions(IProgressMonitor pm)
 			return status;
 		}
 		try {
-			CompletableFuture<List<LanguageServer>> serverList = LanguageServiceAccessor.getLanguageServers(document, LSPRenameProcessor::isPrepareRenameProvider) ;
-			for (LanguageServer serverToTry : serverList.get(500, TimeUnit.MILLISECONDS)) {
-				// check if prepareRename is supported by the active LSP
-				if (languageServer.equals(serverToTry)) {
-					final var identifier = LSPEclipseUtils.toTextDocumentIdentifier(document);
-					final var params = new PrepareRenameParams();
-					params.setTextDocument(identifier);
-					params.setPosition(LSPEclipseUtils.toPosition(offset, document));
-					prepareRenameResult = languageServer.getTextDocumentService().prepareRename(params).get(1000, TimeUnit.MILLISECONDS);
-					if (prepareRenameResult == null) {
-						status.addFatalError(Messages.rename_invalidated);
-					}
-				}
+			final var identifier = LSPEclipseUtils.toTextDocumentIdentifier(document);
+			final var params = new PrepareRenameParams();
+			params.setTextDocument(identifier);
+			params.setPosition(LSPEclipseUtils.toPosition(offset, document));
+
+
+			Optional<Pair<LanguageServerWrapper, Either3<Range, PrepareRenameResult, PrepareRenameDefaultBehavior>>> tmp = initialExecutor.withFilter(LSPRenameProcessor::isPrepareRenameProvider)

```

Comment:
I understand that this works, but I do not quite like to change executors like this. Do you think is it a good pattern to promote? I would rather use LanguageServers.forDocument(document).withFilter(...) and do not pass the executor to the method at all. We already pass the document and reusing the executor does not bring anything here. What do you think?

Reply:
You're right. Passing an executor and then changing the filter amounts to no more than passing the document in a needlessly elaborate way...